### DISCUSSION TITLE
Add backlinks to GET vault/:file response

### DISCUSSION THREAD

**Discussion by AlgusDark** [coddingtonbear/obsidian-local-rest-api#41] [Ideas]
Thanks for this awesome project.  I believe that adding backlinks in the response of this endpoint would be a nice addition. I didn't have too much time looking at the code, but maybe the change for that is in the [getFileMetadataObject](https://github.com/coddingtonbear/obsidian-local-rest-api/blob/64f04f4f4efed9f872e5816304381decc60c1c0d/src/requestHandler.ts#LL105C9-L105C30) function adding this:

```ts
interface BacklinkMetadata{
  displayText: string
  link: string
  original: string
  position: { start: any, end: any }
}

interface BacklinksResponse{
  data: {
    [filename: string]: Array<BacklinkMetadata>
  }
}
 
const backlinks: BacklinksResponse = this.app.metadataCache.getBacklinksForFile(file);
```

**Comment by coddingtonbear** [comment]
I think this is an excellent idea, @AlgusDark!  If you'd be willing to post a pull request adding support for that, I'd be glad to work with you on getting that feature added.

**Comment by avigilante** [comment]
@AlgusDark / @coddingtonbear Has anything ever happened re backlinks?
I've built a local AI stack (Qdrant, Llama, my own MCP) that works on top of my vault, far more powerful than the standard Obsidian MCP. The only function that I'm missing (and for which I had to build a workaround) is backlinks.
Given Obsidian's new terminal UI - which does offer Backlinks function - I'm considering extending my MCP so that it can work with Obsidian terminal UI, but I'd only use it for backlinks...

In any case, THANK YOU for this plugin. None of the AI stack I've built would have ever been possible without you.

**Comment by coddingtonbear** [comment]
No progress that I've heard of, unfortunately, @avigilante.  If you're up for submitting a patch, feel free -- just make sure to follow the contribution guidelines (https://github.com/coddingtonbear/obsidian-local-rest-api/blob/main/CONTRIBUTING.md).

**Comment by avigilante** [comment]
Thanks! I've actually solved backlinks through the new Obsidian CLI.
So, now my MCP uses your REST API for everything else but backlinks.

**Comment by coddingtonbear** [comment]
I'm going to track this as #247

**Comment by coddingtonbear** [comment]
& backlinks are now in the metadata as of v4.0.0!
